yt_dlp/extractor/cybrary.py

```diff
@@ -0,0 +1,147 @@
+﻿# coding: utf-8
+from .common import InfoExtractor
+
+from ..utils import (
+    ExtractorError,
+    smuggle_url,
+    str_or_none,
+    traverse_obj,
+    urlencode_postdata
+)
+
+
+class CybraryBaseIE(InfoExtractor):
+    _API_KEY = 'AIzaSyCX9ru6j70PX2My1Eq6Q1zoMAhuTdXlzSw'
+    _ENDPOINTS = {
+        'course': 'https://app.cybrary.it/courses/api/catalog/browse/course/{}',
+        'course_enrollment': 'https://app.cybrary.it/courses/api/catalog/{}/enrollment',
+        'enrollment': 'https://app.cybrary.it/courses/api/enrollment/{}',
+        'launch': 'https://app.cybrary.it/courses/api/catalog/{}/launch',
+        'vimeo_oembed': 'https://vimeo.com/api/oembed.json?url=https://vimeo.com/{}',
+    }
+    _NETRC_MACHINE = 'cybrary'
+    _TOKEN = None
+
+    def _perform_login(self, username, password):
+        response = self._download_json(
+            f'https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key={self._API_KEY}', None, note='Logging in',
+            data=urlencode_postdata({'email': username, 'password': password, 'returnSecureToken': True}))
+
+        if response:
+            self._TOKEN = response['idToken']
+
```

Comment:
since login is mandatory, add this:
```suggestion

    def _real_initialize(self):
        if not self._TOKEN:
            self.raise_login_required(method='password')

```